dive_core/CMakeLists.txt

```diff
@@ -85,20 +117,27 @@ endif()
 file(GLOB_RECURSE COMMON_HDR_FILES "${CMAKE_SOURCE_DIR}/dive_core/common/*.h")
 file(GLOB_RECURSE COMMON_SRC_FILES "${CMAKE_SOURCE_DIR}/dive_core/common/*.cpp")
 
+file(GLOB_RECURSE FREEDRENO_SRC_FILES "${THIRDPARTY_DIRECTORY}/mesa/src/freedreno/common/freedreno_dev_info.c")
```

Comment:
For my info and out of curiosity... you did all this modification to CMakeLists.txt in order to support what? Building a header or .c file? And why is it necessary to have that header and/or source file to deal with variants?

Reply:
to call fd_dev_info to get gpu id based on chip id defined in mesa\src\freedreno\common\freedreno_devices.py

Reply:
And I take it fd_dev_info is non-trivial code that is best not duplicated within Dive, right?

Reply:
it is using generated header from python, my worry is that if we duplicate the code, we need to track chip id of new devices. also the code is in c which violates some cpp rules. :( so simpler compiling their code instead of duplicating in Dive.

Reply:
Okie dokie, just checking!